src/audio/copier/copier.c

```diff
@@ -617,9 +619,10 @@ static int copier_prepare(struct comp_dev *dev)
 		}
 	} else {
 		/* set up format conversion function */
-		cd->converter[0] = get_converter_func(&cd->config.base.audio_fmt,
-						      &cd->config.out_fmt, ipc4_gtw_none,
-						      ipc4_bidirection);
+		for (int i = 0; i < IPC4_COPIER_MODULE_OUTPUT_PINS_COUNT; i++)
+			cd->converter[i] = get_converter_func(&cd->config.base.audio_fmt,
+							      &cd->config.out_fmt, ipc4_gtw_none,
+							      ipc4_bidirection);
```

Comment:
`get_converter_func()`'s parameters don't change with the loop iterations, so, the result will always be the same? Then maybe better keep the above line for `cd->converter[0]` then the below check and after that in an `else` branch copy from index 0 to all other array members?

Reply:
Fixed